public/app/core/utils/richHistory.ts

```diff
@@ -329,6 +329,54 @@ export function notEmptyQuery(query: DataQuery) {
   return false;
 }
 
+export function filterQueriesBySearchFilter(queries: RichHistoryQuery[], searchFilter: string) {
+  return queries.filter(query => {
+    const hasMatchingComment = query.comment.includes(searchFilter);
+    if (hasMatchingComment) {
+      return true;
+    }
+
+    const hasMatchingQueries = query.queries.filter(query =>
+      // Remove fields in which we don't want to be searching
+      Object.values(_.omit(query, ['datasource', 'key', 'refId'])).some(value =>
+        value.toString().includes(searchFilter)
+      )
+    );
+
+    return hasMatchingQueries.length > 0;
```

Comment:
`hasMatchingQueries` actually isn't a boolean but an array. Besides, you don't even need it. Unless you want to argue that you find it more readable when you split up into filtering and the return expression. In which case I'd say "Alright but then please rename it." ;) (But you definitely don't need `hasMatchingComment`.)

Reply:
I have renamed it to `listOfMatchingQueries`. I have removed hasMatchingComment.